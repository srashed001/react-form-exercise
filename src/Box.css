.Box {
    display: inline-block;
    margin: 10px;

}