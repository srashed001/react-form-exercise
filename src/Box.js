
import './Box.css'

const Box = ({ color = 'red', width = 100, height=100}) => {
    const style = {
        backgroundColor: color, 
        width: `${width}px`,
        height: `${height}px`
    }
    return (
        <div 
        className="Box" 
        style={style}
        data-testid="box"
        >

        </div>
    )

}

export default Box