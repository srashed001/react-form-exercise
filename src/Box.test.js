import { render } from '@testing-library/react'
import Box from './Box'

it ("renders without crashing", function(){
    render(<Box/>)
})

it("matches snapshot", function(){
    const {asFragment} = render(<Box />)
    expect(asFragment()).toMatchSnapshot();
})

it("matches snapshot", function(){
    const {asFragment} = render(<Box />)
    expect(asFragment()).toMatchSnapshot();
})

it("creates styles according to props", function(){
    const {getByTestId} = render(<Box  color='red' width={100} height={100}/>)
    const div = getByTestId('box');
    expect(div).toHaveAttribute('style', "background-color: red; width: 100px; height: 100px;")

})