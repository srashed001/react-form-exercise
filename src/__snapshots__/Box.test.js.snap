// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    data-testid="box"
    style="background-color: red; width: 100px; height: 100px;"
  />
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="Box"
    data-testid="box"
    style="background-color: red; width: 100px; height: 100px;"
  />
</DocumentFragment>
`;
